plugins/source/github/client/client.go

```diff
@@ -91,10 +93,15 @@ func New(ctx context.Context, logger zerolog.Logger, spec Spec) (schema.ClientMe
 		if err != nil {
 			return nil, fmt.Errorf("failed to parse InstallationID %v: %w", auth.InstallationID, err)
 		}
+		transport := http.DefaultTransport
+		if spec.LocalCachePath != "" {
+			cache := diskcache.New(spec.LocalCachePath)
```

Comment:
Do we have to validate that `spec.LocalCachePath` is a directory or file?

Reply:
Done in https://github.com/cloudquery/cloudquery/pull/17368/commits/03203fa62649e3728d39f0840958f90799f2f485

The cache package already does `os. MkdirAll` which should throw a relevant error, but that we might want to fail early here instead of on first write to the cache